lib/SIL/AbstractionPattern.cpp

```diff
@@ -341,6 +358,25 @@ AbstractionPattern::getCFunctionAsMethodSelfPattern(CanType selfType) const {
                            getGenericSignatureForFunctionComponent(), selfType);
 }
 
+AbstractionPattern
+AbstractionPattern::getCXXMethodSelfPattern(CanType selfType) const {
+  assert(hasStoredCXXMethod());
+  auto CXXMethod = getCXXMethod();
+  if (CXXMethod->isInstance()) {
+    // Use the clang type for the receiver type.  If this is ever
+    // insufficient --- if we have interesting bridging to do to
+    // 'self' --- we have the right information to be more exact.
+    auto clangSelfType =
+        CXXMethod->getThisType(CXXMethod->getASTContext()).getTypePtr();
+    return AbstractionPattern(getGenericSignatureForFunctionComponent(),
+                              selfType, clangSelfType);
+  }
+  // The formal metatype parameter to a C function imported as a static method
+  // is dropped on the floor. Leave it untransformed.
```

Comment:
The basics of this comment are still right, but please update it to talk about C++ `static` methods.